presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -248,6 +248,30 @@ public void testDateFilter()
         }
     }
 
+    @Test
+    public void testJoinsWithConstantFilterAndNoCriteria()
+    {
+        // Test double filtered left join with right constant equality.
+        assertQuery("SELECT COUNT(*) FROM (SELECT * FROM lineitem WHERE orderkey % 1024 = 0) "
```

Comment:
Did you try any cases where the right side (of left join) and left side (of right join) are empty tables so that we can check the OUTER JOIN semantics ?